src/coreclr/jit/valuenum.cpp

```diff
@@ -4714,6 +4714,26 @@ ValueNum ValueNumStore::EvalUsingMathIdentity(var_types typ, VNFunc func, ValueN
                 {
                     resultVN = VNOneForType(typ);
                 }
+                else if (varTypeIsIntegralOrI(TypeOfVN(arg0VN)))
+                {
+                    ZeroVN = VNZeroForType(typ);
+                    if (genTreeOps(func) == GT_GE)
```

Comment:
Can we handle VNF_GE_UN as well?

Reply:
Also, do you handle `NN < 0` => false? (and its reversed version)

Reply:
> Also, do you handle NN < 0 => false? (and its reversed version)
Yes, I handle it below.

Reply:
> Can we handle VNF_GE_UN as well?

What does `genTreeOps(VNF_GE_UN)` turn into? Would it just be `GT_GE` ?

Reply:
> > Can we handle VNF_GE_UN as well?
> 
> What does `genTreeOps(VNF_GE_UN)` turn into? Would it just be `GT_GE` ?

It doesn't map into GenTreeOps, just check `func == VNF_GE_UN`

Reply:
Gotcha, then adding the checks should be good then.

Reply:
So it looks like we already handle `VNF_GE_UN` and others:
https://github.com/dotnet/runtime/blob/d6665c28dcbf7c71ec28969eb6382a8c6ade8f39/src/coreclr/jit/valuenum.cpp#L4852
https://github.com/dotnet/runtime/blob/d6665c28dcbf7c71ec28969eb6382a8c6ade8f39/src/coreclr/jit/valuenum.cpp#L4869

Therefore, we don't need to add additional checks for them.